web-registry/src/pages/ProjectEdit/ProjectEdit.tsx

```diff
@@ -64,6 +67,7 @@ function ProjectEdit(): JSX.Element {
   const [isWarningModalOpen, setIsWarningModalOpen] = useState<
     string | undefined
   >(undefined);
+  const [dirty, setDirty] = useState<boolean>(false);
```

Comment:
I did something similar [here](https://github.com/regen-network/regen-web/pull/1884/files#diff-900e135612e75db44f8111fa517d075324ba01a10c3cfcc28cf029c028aeaf96R141) but with a ref, since we don't necessarily want to trigger a rerender when updating this state but rather only check it while performing a navigation/back action.

Reply:
good point, will update!

Reply:
8adbbde51da97345523546e5661bd84c8f6edae0